### PR TITLE
Use extended mediarecorder

### PR DESCRIPTION
**Pull request by marcoambrosini**
Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

File: lib/Controller/PageController.php [hunk 1]
```diff
@@ -262,6 +262,8 @@ public function index(string $token = '', string $callUser = '', string $passwor
 		$csp = new ContentSecurityPolicy();
 		$csp->addAllowedConnectDomain('*');
 		$csp->addAllowedMediaDomain('blob:');
+		$csp->addAllowedWorkerSrcDomain('blob:');
+		$csp->addAllowedWorkerSrcDomain("'self'");
```
Comment (nickvergessen) [inline lib/Controller/PageController.php:266]: Also needs to go into OCA\Talk\Listener\CSPListener

File: webpack.js [hunk 2]
```diff
@@ -72,4 +73,35 @@ webpackConfig.optimization.minimizer = [
 	}),
 ]
 
+webpackConfig.plugins.push(
+	new WorkboxPlugin.GenerateSW({
+		swDest: 'photos-service-worker.js',
```
Comment (nickvergessen) [inline webpack.js:78]: file no exist

File: webpack.js [hunk 3]
```diff
@@ -72,4 +73,35 @@ webpackConfig.optimization.minimizer = [
...
+		clientsClaim: true,
+		skipWaiting: true,
+		exclude: [new RegExp('.*')], // don't do precaching
+		inlineWorkboxRuntime: true,
+		sourcemap: false,
+
+		// Define runtime caching rules.
+		runtimeCaching: [{
+			// Match any preview file request
+			urlPattern: /^.*\/core\/preview\?fileId=.*/,
```
Comment (nickvergessen) [inline webpack.js:88]: url pattern is not what we interact with so remove

File: webpack.js [hunk 4]
```diff
@@ -72,4 +73,35 @@ webpackConfig.optimization.minimizer = [
...
+
+			// Apply a strategy.
+			handler: 'CacheFirst',
+
+			options: {
+				// Use a custom cache name.
+				cacheName: 'images',
```
Comment (nickvergessen) [inline webpack.js:95]: we dont do images

**Review by Ivansss — APPROVED** [review]
Tested and works 👍

**Pull request merged**